### PR TITLE
[HM3D]--Semantic scene support/bugfixes.

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR adds various functions and refactors required to implement support for HM3D semantic scenes in the correct format.  Included are : 
- Better vertex color handling in semantic mesh load
- More rigorous string verification for sentinel string denoting Semantic Scene Descriptor file type
- Proper application of frame re-orientation to Semantic Mesh data.  This is required to enable any semantic meshes to be re-oriented in engine.
- Force all semantic meshes to be treated as Instance Meshes (temporary so that vertex colors and vertex ids are supported).

Also, some refactors to improve efficiency and debug message content in Resource Manager are included.

## How Has This Been Tested
Locally c++ and python tests

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jturner65** [comment]
@aclegg The tests in install_and_test_ubuntu will not pass until the mp3d example dataset is updated to have and use the scenedataset config.

File: src/esp/assets/ResourceManager.cpp [hunk 1]
```diff
@@ -123,16 +124,26 @@ ResourceManager::~ResourceManager() {
 }
 
 void ResourceManager::buildImporters() {
+  // Preferred plugins, Basis target GPU format
+  importerManager_.setPreferredPlugins("GltfImporter", {"CgltfImporter"});
+#ifdef ESP_BUILD_ASSIMP_SUPPORT
+  importerManager_.setPreferredPlugins("ObjImporter", {"AssimpImporter"});
+  Cr::PluginManager::PluginMetadata* const assimpmetadata =
+      importerManager_.metadata("AssimpImporter");
+  assimpmetadata->configuration().setValue("ImportColladaIgnoreUpDirection",
```
Comment (Skylion007) [inline src/esp/assets/ResourceManager.cpp:133]: Hmm, I wish there was a way to use this if it wasn't specified in the JSON...

Comment (jturner65) [reply src/esp/assets/ResourceManager.cpp:133]: Could you explain what you mean? Thanks :)

Comment (Skylion007) [reply src/esp/assets/ResourceManager.cpp:133]: I am worried the fact that other importers like Blender may support Collada's up direction may cause confusion for end-users when the rotations do not match. Seems like it's harmful to ignore an up direction defined in the dae. What about other formats that have their own up direction, it feels like we should try to be consistent with the handling of the up direction across formats. (For instance, the rotation in the dataset should be relative to the up direction in the mesh file).

**Review by aclegg3 — APPROVED** [review]
LGTM

**Pull request merged**